src/main/java/com/xiaomi/infra/pegasus/rpc/async/TableHandler.java

```diff
@@ -118,22 +122,36 @@ void initTableConfiguration(query_cfg_response resp) {
       newConfig.replicas.add(newReplicaConfig);
     }
 
-    // set partition configuration by resp, and create sessions
+    // create sessions for primary and secondaries
     FutureGroup<Void> futureGroup = new FutureGroup<>(resp.getPartition_count());
     for (partition_configuration pc : resp.getPartitions()) {
       ReplicaConfiguration s = newConfig.replicas.get(pc.getPid().get_pidx());
       s.ballot = pc.ballot;
-      s.primary = pc.primary;
-      if (pc.primary.isInvalid()) {
-        s.session = null;
-      } else {
-        if (s.session == null || !s.session.getAddress().equals(pc.primary)) {
-          // reset to new primary
-          s.session = manager_.getReplicaSession(pc.primary);
-          ChannelFuture fut = s.session.tryConnect();
-          if (fut != null) {
-            futureGroup.add(fut);
-          }
+
+      // create primary sessions
+      s.primaryAddress = pc.primary;
+      if (!pc.primary.isInvalid()) {
+        s.primarySession = manager_.getReplicaSession(pc.primary);
+        ChannelFuture fut = s.primarySession.tryConnect();
+        if (fut != null) {
+          futureGroup.add(fut);
+        }
+
+        // backup request is enabled, get all secondary sessions
+        s.secondarySessions.clear();
+        if (isEnableBackupRequest()) {
+          // secondary sessions
+          pc.secondaries.forEach(
+              secondary -> {
+                if (!secondary.isInvalid()) {
+                  ReplicaSession session = manager_.getReplicaSession(secondary);
+                  s.secondarySessions.add(session);
+                  ChannelFuture channelFuture = session.tryConnect();
```

Comment:
In the same scope, we cannot define same same for two variables.
And we cann`t use one variable for them, because variable in java is reference to a value, just like `ChannelFuture &fut` in c++